File: Components/ASP.NET/CS/ASPxSchedulerDemos/UserControls/DemoTabs.ascx.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;
using DevExpress.Web.Demos;

public partial class UserControls_DemoTabs : System.Web.UI.UserControl {
    protected void Page_Load(object sender, EventArgs e) {
        var index = 0;
        if(Utils.CurrentDemoPage != null) {
            foreach(SourceCodePage page in Utils.GetCurrentSourceCodePages()) {
                var tab = DemoPageControl.TabPages.Add(page.Title, page.Title);
                var codeViewer = (CodeViewerControl)LoadControl("~\\UserControls\\CodeViewer.ascx");
                codeViewer.SourceCodePage = page;
                codeViewer.Index = index;
                DemosModel currentModel = DemosModel.Instance;
                if (currentModel != null)
                    codeViewer.DisableTextWrap = DemosModel.Instance.DisableTextWrap;
                tab.Controls.Add(codeViewer);
                index++;
            }
        }
    }
}

File: Components/ASP.NET/VB/ASPxNavigationAndLayoutDemos/UserControls/DemoTabs.ascx.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports DevExpress.Web.Demos

Partial Public Class UserControls_DemoTabs
	Inherits System.Web.UI.UserControl
	Protected Sub Page_Load(ByVal sender As Object, ByVal e As EventArgs)
		Dim index = 0
		If Utils.CurrentDemoPage IsNot Nothing Then
			For Each page As SourceCodePage In Utils.GetCurrentSourceCodePages()
				Dim tab = DemoPageControl.TabPages.Add(page.Title, page.Title)
				Dim codeViewer = CType(LoadControl("~\UserControls\CodeViewer.ascx"), CodeViewerControl)
				codeViewer.SourceCodePage = page
				codeViewer.Index = index
				Dim currentModel As DemosModel = DemosModel.Instance
				If currentModel IsNot Nothing Then
					codeViewer.DisableTextWrap = DemosModel.Instance.DisableTextWrap
				End If
				tab.Controls.Add(codeViewer)
				index += 1
			Next page
		End If
	End Sub
End Class
